shop3/looping-tasks/loop-extensions.lisp

```diff
@@ -93,10 +92,75 @@
                                      unifier
                                      ))))
 
+(defun record-reductions (ess-search-state reductions)
+  (when *enhanced-plan-tree*
+    (with-slots (backtrack-stack)
+                           ess-search-state
```

Comment:
Kind of a nit, but I really dislike having tab characters in source files.  We had them in a project once, and things went bonkers when a Windows user got involved, because Windows tabs default to *8* characters instead of *4*.  There's a hook you can add to the save-file for lisp mode that will untabify everything.